### PR TITLE
API fetching

### PR DESCRIPTION
**Pull request by BenM4k**
In this update, I: 
- Made use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as my backend.
- Used [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make the HTTP requests to the backend
- Fetched my books from the Bookstore API after the component has rendered
- Refactored the add book and remove book features to persist the changes in the server

### CODE REVIEW

File: src/components/BookList.jsx [hunk 1]
```diff
@@ -1,69 +1,56 @@
...
 import React from 'react';
-import PropTypes from 'prop-types';
 import { deleteBook } from '../redux/slices/books/bookSlice';
 
-function BookList({ books }) {
+function BookList() {
+  const {
+    books, isLoading, errorMsg,
+  } = useSelector((store) => store.book);
   const dispatch = useDispatch();
```
Comment (Zuhaib042) [inline src/components/BookList.jsx:9]: - [ ] Right now your app is not working as expected and whenever I try to add a book or remove it, I need to refresh the [age to see the difference. You may need to change your approach in order to meet the requirements. Kindly consider importing the `getBooks` action creator here from booksSlice and then also consider importing useEffect hook and then calling that action creator in the useEffect hook.

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @BenM4k 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✅ Descriptive PR
✅ Professional Readme

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @BenM4k ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rotimiazeez — COMMENTED** [review]
# Hi @BenM4k, 

### While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project that has been merged. Kindly ensure your PR are always open with the necessary changes present when submitting for review 👍🏽 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

**Review by franclobo — COMMENTED** [review]
Hi @BenM4k,

# STATUS: APPROVED 😃  

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Correct use of API ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**